ingestion/src/metadata/ingestion/source/pipeline/gluepipeline/metadata.py

```diff
@@ -145,9 +180,95 @@ def get_downstream_tasks(self, task_unique_id, tasks):
                 downstream_tasks.append(self.task_id_mapping[edges["DestinationId"]])
         return downstream_tasks
 
+    def get_lineage_details(self, job) -> Optional[dict]:
+        """
+        Get the Lineage Details of the pipeline
+        """
+        lineage_details = {"sources": [], "targets": []}
+        try:
+            job_details = JobNodeResponse.model_validate(
+                self.glue.get_job(JobName=job)
+            ).Job
+            if job_details and job_details.config_nodes:
+                nodes = job_details.config_nodes
+                for _, node in nodes.items():
+                    for key, entity in node.items():
+                        table_model, storage_model = None, None
+                        if key in TABLE_MODEL_MAP:
+                            table_model = TABLE_MODEL_MAP[key].model_validate(entity)
+                        elif "Catalog" in key:
+                            table_model = CatalogSource.model_validate(entity)
+                        elif key in STORAGE_MODEL_MAP:
+                            storage_model = STORAGE_MODEL_MAP[key].model_validate(
+                                entity
+                            )
+                        if table_model:
+                            for db_service_name in self.get_db_service_names():
+                                table_entity = self.metadata.get_by_name(
+                                    entity=Table,
+                                    fqn=fqn.build(
+                                        metadata=self.metadata,
+                                        entity_type=Table,
+                                        table_name=table_model.table_name,
+                                        database_name=table_model.database_name,
+                                        schema_name=table_model.schema_name,
+                                        service_name=db_service_name,
+                                    ),
+                                )
+                                if table_entity:
+                                    if key.endswith("Source"):
+                                        lineage_details["sources"].append(
+                                            (table_entity, "table")
+                                        )
+                                    else:
+                                        lineage_details["targets"].append(
+                                            (table_entity, "table")
+                                        )
+                                    break
+                        if storage_model:
+                            for (
+                                storage_service_name
+                            ) in self.get_storage_service_names():
+                                for path in storage_model.Paths or [storage_model.Path]:
+                                    storage_entity = self.metadata.get_by_name(
```

Comment:
also a better way to get the container is [es_search_container_by_path](https://github.com/open-metadata/OpenMetadata/blob/53cdd34391e5a541d153d89786dafcb65365e900/ingestion/src/metadata/ingestion/ometa/mixins/es_mixin.py#L192)

where we pass the full s3 path and get the entity in return, instead of manipulating or breaking the path and trying to figure out fqn